doc/source/whatsnew/v0.20.0.txt

```diff
@@ -44,6 +44,18 @@ fixed-width text files, and :func:`read_excel` for parsing Excel files.
    pd.read_fwf(StringIO(data)).dtypes
    pd.read_fwf(StringIO(data), dtype={'a':'float64', 'b':'object'}).dtypes
 
+Reading dataframes from URLs, in :func:`read_csv` or :func:`read_table`, now
+supports additional compression methods (`xz`, `bz2`, `zip`). Previously, only
+`gzip` compression was supported. By default, compression of URLs and paths are
+now both inferred using their file extensions.
+
+.. ipython:: python
+
+   url = ('https://github.com/pandas-dev/pandas/raw/master/' +
+          'pandas/io/tests/parser/data/salaries.csv.bz2')
+   pd.read_table(url, compression='infer')  # default, infer compression
+   pd.read_table(url, compression='xz')  # explicitly specify compression
```

Comment:
Didn't realize this code would get executed. Now assigns to `df` and then uses head to display first two rows. In https://github.com/pandas-dev/pandas/pull/14880/commits/cb40a41bd26ef50aedb3923235d648cd4cd4aa94.